modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php

```diff
@@ -0,0 +1,240 @@
+
+      $anchor = \Drupal::service('focal_point.manager')
+        ->absoluteToRelative($crop->x->value, $crop->y->value, $width, $height);
+      $focal_point_default_value = "{$anchor['x']},{$anchor['y']}";
+
+      $form['focal'] = [
+        '#type' => 'fieldset',
+        '#title' => $this->t('Set focal Point'),
+      ];
+      // Should be unique class for textfield.
+      $focal_point_selector = 'focal-point-' . $style->id();
+
+      $form['focal'][$style->id()]['indicator'] = [
+        '#type' => 'html_tag',
+        '#tag' => 'div',
+        '#attributes' => [
+          'class' => ['focal-point-indicator'],
+          'data-selector' => $focal_point_selector,
+//          'data-delta' => $delta,
+        ],
+      ];
+
+      $form['focal'][$style->id()]['preview'] = [
+        '#theme' => 'image_style',
+        '#width' => $width,
+        '#height' => $height,
+        // @TODO: Avoid hardcoded style name. It comes from field settings.
+        '#style_name' => 'thumbnail_focal_point',
+        '#uri' => $file->getFileUri(),
+      ];
+
+      $form['focal'][$style->id()]['focal_point'] = [
+        '#type' => 'textfield',
+        '#title' => $this->t('Focal point'),
+        '#description' => $this->t('Specify the focus of this image in the form "leftoffset,topoffset" where offsets are in percents. Ex: 25,75'),
+        '#default_value' => $focal_point_default_value,
+        '#attributes' => [
+          'class' => ['focal-point', $focal_point_selector],
+          'data-selector' => $focal_point_selector,
+        ],
+        '#wrapper_attributes' => [
+          'class' => ['focal-point-wrapper', 'visually-hidden', 'hidden'],
+        ],
+        '#attached' => [
+          'library' => ['focal_point/drupal.focal_point'],
+        ],
+      ];
+    }
+
+    $form['save_focal_point'] = [
+      '#type' => 'submit',
+      '#name' => 'save_focal_point',
+      '#op' => 'save_focal_point',
+      '#value' => $this->t('Save Focal Point'),
+      '#ajax' => [
+        'callback' => '::ajaxSave',
+      ],
+    ];
+
+    $form['close_dialog'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Close'),
+      '#ajax' => [
+        'callback' => '::closePopup',
+      ],
+    ];
+
+    $form['#attached']['library'][] = 'openy_focal_point/openy_focal_point';
+
+    return $form;
+  }
+
+  public static function ajaxSave(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\file\Entity\File $file */
+    $file = $form_state->get('file');
+
+    $crop_type = \Drupal::config('focal_point.settings')->get('crop_type');
+    $crop = Crop::findCrop($file->getFileUri(), $crop_type);
+
+    /** @var \Drupal\focal_point\FocalPointManagerInterface $focal_point_manager */
+    $focal_point_manager = \Drupal::service('focal_point.manager');
+    list($x, $y) = explode(',', $form_state->getValue('focal_point'));
+    $image = \Drupal::service('image.factory')->get($file->getFileUri());
```

Comment:
we should pass service via DI

Reply:
we use this service in public static method, so we don't have instance of the form. This is why we can not use DI here.